### PR TITLE
Add converter from IEnumerable to TheoryData

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Manipulating test input data in IEnumerable format is
desirable, but a bit awkward to get into IEnumerable<object[]> format that [Theory] expects.

Add test project to new test project. :)  Change a test in
System.IO.FileSystem to theory with new extension.

### CODE REVIEW

File: src/CoreFx.Private.TestUtilities/src/System/TheoryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+using Xunit;
+
+namespace System
+{
+    public static class TheoryExtensions
+    {
+        /// <summary>
+        /// Converts an IEnumerable<typeparamref name="T"/> into an Xunit theory compatible enumerable.
+        /// </summary>
+        public static TheoryData ToTheoryData<T>(this IEnumerable<T> data)
```
Comment (tarekgh) [inline src/CoreFx.Private.TestUtilities/src/System/TheoryExtensions.cs]: >ToTheoryData< [](start = 33, length = 13)

is it possible we add this to the xunit extensions instead?

[https://github.com/dotnet/buildtools/tree/master/src/xunit.netcore.extensions](https://github.com/dotnet/buildtools/tree/master/src/xunit.netcore.extensions)

File: src/CoreFx.Private.TestUtilities/tests/TheoryExtensionTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace CoreFx.Private.TestUtilities.Tests
```
Comment (tarekgh) [inline src/CoreFx.Private.TestUtilities/tests/TheoryExtensionTests.cs]: >CoreFx.Private.TestUtilities.Tests [](start = 10, length = 34)

interesting idea to add tests to this test library :-)

**Comment by tarekgh** [comment]
CC @weshaggard

File: src/CoreFx.Private.TestUtilities/src/System/TheoryExtensions.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/src/System/TheoryExtensions.cs]: I think we should migrate things there eventually. Let them bake here a bit and move them up (same goes for our Assert extensions, etc.).

File: src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln [hunk 3]
```diff
@@ -13,26 +13,119 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{E107E9C1-E89
...
+		uapaot-Windows_NT-Release|Any CPU = uapaot-Windows_NT-Release|Any CPU
+		uap-Windows_NT-Debug|Any CPU = uap-Windows_NT-Debug|Any CPU
+		uap-Windows_NT-Release|Any CPU = uap-Windows_NT-Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{5B7EAEC-93CB-40DF-BE40-A60BC189B737}.Debug|Any CPU.ActiveCfg = netcoreapp-Windows_NT-Debug|Any CPU
-		{5B7EAEC-93CB-40DF-BE40-A60BC189B737}.Debug|Any CPU.Build.0 = netcoreapp-Windows_NT-Debug|Any CPU
-		{5B7EAEC-93CB-40DF-BE40-A60BC189B737}.Release|Any CPU.ActiveCfg = netcoreapp-Windows_NT-Release|Any CPU
-		{5B7EAEC-93CB-40DF-BE40-A60BC189B737}.Release|Any CPU.Build.0 = netcoreapp-Windows_NT-Release|Any CPU
+		{EBDB0247-CA43-4226-B7A1-8FEF21061D09}.Debug|Any CPU.ActiveCfg = netfx-Windows_NT-Debug|Any CPU
```
Comment (weshaggard) [inline src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln]: Did you run the UpdateVSConfigurations target? I wouldn't expect all these in the solution after that update.

Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln]: No, I'll do that- what should I run it against?

File: referenceFromRuntime.targets [hunk 4]
```diff
@@ -12,8 +12,8 @@
   <Target Name="AddRuntimeProjectReference"
           BeforeTargets="AddProjectReferencesDynamically"
           Condition="'$(IsTestProject)'!='true' AND '@(ReferenceFromRuntime)' != ''">
-    <Error Condition="'$(IsReferenceAssembly)' != 'true' AND '$(RuntimeProjectFile)' == ''" Text="RuntimeProjectFile must be specified when using ReferenceFromRuntime from source projects." />
-    <Error Condition="'$(IsReferenceAssembly)' == 'true'" Text="ReferenceFromRuntime may not be used from reference assemblies." />
+    <Error Condition="('$(IsReferenceAssembly)' != 'true' OR '$(AllowReferenceFromRuntime)' == 'true') AND '$(RuntimeProjectFile)' == ''" Text="RuntimeProjectFile must be specified when using ReferenceFromRuntime from source projects." />
```
Comment (weshaggard) [inline referenceFromRuntime.targets]: I think I would add the condition to the Target itself.

File: src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj [hunk 5]
```diff
@@ -26,6 +31,9 @@
     <Compile Include="$(CommonTestPath)\System\IO\PathFeatures.cs">
       <Link>Common\System\IO\PathFeatures.cs</Link>
     </Compile>
+    <Compile Include="$(CommonTestPath)\System\PlatformDetection.cs">
```
Comment (weshaggard) [inline src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj]: orthogonal but I want PlatformDetection to be in the TestUtilities as well at some point.

Comment (tarekgh) [inline src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj]: @JeremyKuhne talked offline with me and we have same idea.

**Comment by tarekgh** [comment]
LGTM.

File: src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln (hunk 3 above)
Comment (weshaggard) [inline src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln]: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#updating-configurations

File: referenceFromRuntime.targets (hunk 4 above)
Comment (JeremyKuhne) [inline referenceFromRuntime.targets]: How would I do that? I need to run the target without throwing these errors.

File: src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln (hunk 3 above)
Comment (JeremyKuhne) [inline src/CoreFx.Private.TestUtilities/CoreFx.Private.TestUtilities.sln]: Thanks :)

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows x86 Release Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot Windows x64 Debug Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows x64 Debug Build

File: src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj (hunk 5 above)
Comment (hughbe) [inline src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj]: What about AssertExtensions?

Comment (tarekgh) [inline src/System.IO.FileSystem/tests/Performance/System.IO.FileSystem.Performance.Tests.csproj]: @hughbe any test common code should be moved to the TestUtilities library. are you interested doing that? that will be fantastic.

**Pull request merged**